File: apps/codemap-search/tests/fixtures/navigation/java/worker.java
import org.springframework.scheduling.annotation.Scheduled;

class CheckoutWorker {
    private final JobQueue queue;
    private final JobProcessor processor;
    private final Logger logger;

    @Scheduled(fixedDelay = 1000)
    void tick() {
        Job job = queue.next();
        if (job == null) {
            logger.warn("no job");
            return;
        }

        Receipt receipt = processor.process(job);
        queue.ack(receipt.id());
    }
}

File: apps/codemap-search/tests/fixtures/navigation/kotlin/worker.kt
import org.springframework.scheduling.annotation.Scheduled

class CheckoutWorker(
    private val queue: JobQueue,
    private val processor: JobProcessor,
    private val logger: Logger,
) {
    @Scheduled(fixedDelay = 1000)
    fun tick() {
        val job = queue.next()
        if (job == null) {
            logger.warn("no job")
            return
        }

        val receipt = processor.process(job)
        queue.ack(receipt.id)
    }
}
